src/dparse/ast.d

```diff
@@ -3188,6 +3201,7 @@ final class TernaryExpression : ExpressionNode
 }
 
 ///
+// deprecated("To be replaced by ExpressionStatement + ThrowExpression")
```

Comment:
Not sure what's the proper deprecation process is. 
Couldn't mark `ThrowStatement` as deprecated due to [this bug](https://issues.dlang.org/show_bug.cgi?id=22668).

Reply:
I think it might be best to remove/rename it completely (causing compiler errors) and have the new behavior already and do a version bump. Currently as I see it this change would mean ThrowStatement wouldn't be used at all anymore after some point, causing old code to still compile but no longer work as expected.

Reply:
> [...]  ThrowStatement wouldn't be used at all anymore after some point, causing old code to still compile but no longer work as expected.

The idea was to remove `ThrowStatement` once it's no longer parsed. Any program that has been adapted to  `ThrowExpression` would keep working as expected and could simply remove the `ThrowStatement`-related code.

But outright removing it now should also be fine - assuming that users can likely transition by replacing  `ThrowStatement`. with `ThrowExpression`.

Reply:
if they dependend on ThrowStatement: (in a visitor)
- if we keep it and also add ThrowExpression, the visit function is likely no longer going to be called as expected anymore or have some changes in accessed member properties (potentially causing silent breakage)
- if we remove it and move all the relevant properties into ThrowExpression, the code will no longer compile and they can see that they need to update their code (loud breakage, no runnable wrong code)

So I think removing it is the better choice.

sadly I don't think we can alias ThrowStatement to some broken type to cause additional error messages, which would be optimal in guiding users how to upgrade.

Reply:
The old `ThrowStatement` is now an alias to `ThrowExpression` and the field names are now identical. So existing code referencing `ThrowStatement` should keep working with a deprecation message.